### ISSUE TITLE
Only retry for idempotent methods

### ISSUE THREAD

**Issue by htfy96** [softonic/axios-retry#17]
Current implementation will resend the requests even for `POST` method. However, it could be the case that the request has arrived at the server, while the response timed out. In this situation, this library will resend the `POST` request and the server will then receive a disastrous duplicate `POST`. So I was wondering that is there any way to retry only for idempotent methods?

**Comment by htfy96** [comment]
Here is my workaround. Hope that it would be default configuration:
```js
const retryConfig = {
  retries: 3,
  retryCondition (err) {
    // only retry for idempotent methods
    return ['options', 'get', 'put', 'patch', 'head'].indexOf(err.config.method) > -1
  }
}
```

**Comment by AlanFoster** [comment]
I agree; That's a really simple work-around too :+1:

**Comment by rubennorte** [comment]
@htfy96 have you ever experienced such behaviour? At the moment this plugin just retries safe network connection errors and explicitly avoids timeouts and cancellation errors (see https://github.com/softonic/axios-retry/blob/master/es/index.js#L107).

The `retryCondition` allows you to configure stricter conditions but with the base condition it should be fine.

**Comment by AlanFoster** [comment]
@htfy96 Check out what Stripe do to ensure their endpoints are idempotent:

https://stripe.com/blog/idempotency

**Comment by rubennorte** [comment]
I just released a new version (3.0.0) that retries idempotent requests by default along with network errors.
